paris-material/src/main/java/com/airbnb/paris/material/proxies/MaterialButtonProxy.kt

```diff
@@ -0,0 +1,161 @@
+
+    @Attr(R2.styleable.Paris_MaterialButton_android_textColor)
+    fun setTextColor(colors: ColorStateList?) {
+        this.mColors = colors
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_android_textAppearance)
+    fun setTextAppearance(@StyleRes textAppearance: Int) {
+        mTextAppearance = textAppearance
+    }
+
+
+    @Attr(R2.styleable.Paris_MaterialButton_backgroundTint)
+    fun setBackgroundTintList(tintList:ColorStateList?){
+        view.backgroundTintList = tintList
+
+        Log.d("MaterialProxy", "tintList: $tintList")
+
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_backgroundTintMode)
+    fun setBackgroundTintMode(mode: Int){
+        val tintMode = ViewUtils.parseTintMode(mode, PorterDuff.Mode.SRC_IN)
+        view.backgroundTintMode = tintMode
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_cornerRadius)
+    fun setCornerRadius(@Px cornerRadius: Int) {
+        view.cornerRadius = cornerRadius
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_elevation)
+    fun setElevation(@Px elevation: Int){
+        view.elevation = elevation.toFloat()
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_icon)
+    fun setIcon(icon: Drawable){
+        view.icon = icon
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconSize)
+    fun setIconSize(@Px size: Int){
+        view.iconSize = size
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconGravity)
+    fun setIconGravity(gravity: Int){
+        view.iconGravity = gravity
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconPadding)
+    fun setIconPadding(@Px padding: Int){
+        view.iconPadding = padding
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconTint)
+    fun setIconTintList(tintList:ColorStateList?){
+        view.iconTint = tintList
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconTintMode)
+    fun setIconTintModeMode(mode: Int){
+        val tintMode = ViewUtils.parseTintMode(mode, PorterDuff.Mode.SRC_IN)
+        view.iconTintMode = tintMode
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_rippleColor)
+    fun setRippleColor(ripple:ColorStateList?){
+        view.rippleColor = ripple
+
+        Log.d("MaterialProxy", "ripple: $ripple")
+
+    }
+
+    //todo add shapeAppearance & shapeAppearanceOverlay
+
+    @Attr(R2.styleable.Paris_MaterialButton_strokeColor)
+    fun setStrokeColor(strokeColor: ColorStateList?){
+        view.strokeColor = strokeColor
+
+        Log.d("MaterialProxy", "strokeColor: $strokeColor")
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_strokeWidth)
+    fun setStrokeWidth(@Px width: Int){
+        view.strokeWidth = width
+
+        Log.d("MaterialProxy", "strokeWidth: $width")
+    }
+
+    @AfterStyle
+    fun afterStyle(@Suppress("UNUSED_PARAMETER") style: Style?) {
+        /*
+        * Material component has default textAppearance.
+        * But if we do not pass any textAppearance then
+        * android default textAppearance is set. To keep the
```

Comment:
If the style doesn't define `textAppearance` and the theme does then the theme value will be used. That's probably what you're observing and it is the correct behavior. I don't think we should be trying to override it here.